app/common/elasticsearchclient.py

```diff
@@ -217,6 +217,46 @@ def free_text_search(self, searchphrase,
             data.append(datapoint)
         return PaginatedResult(res, params, data)
 
+
+    def best_hit_search(self,
+                         doc_filter=(FreeTextFilterOptions.ALL),
+                         **kwargs):
+        '''
+       similar to free_text_serach but can take multiple queries
+
+        :param searchphrase:
+        :param doc_filter:
+        :param kwargs:
+        :return:
+        '''
+
+        params = SearchParams(**kwargs)
+        params.q = kwargs.get('q', []) or []
+
+        if doc_filter is None:
+            doc_filter = [FreeTextFilterOptions.ALL]
+        doc_filter = self._get_search_doc_types(doc_filter)
+        results = self._best_hit_query(doc_filter, params)
+        # current_app.logger.debug("Got %d Hits in %ims" % (res['hits']['total'], res['took']))
+        data = []
+        
+        for res in results['responses']:
+            res_data = [];
+            for hit in res['hits']['hits']:
```

Comment:
We agreed to return just the first hit (if there is one). we can return None if there is no match.
If you interate over all res['hits']['hits'] it will return a part of the data in case the size is set >1. this should be a fixed size query returning just the best match